extensions/lifecycle/LifecycleQueuePopulator.js

```diff
@@ -0,0 +1,136 @@
+const { usersBucket, mpuBucketPrefix } = require('arsenal').constants;
+const QueuePopulatorExtension =
+    require('../../lib/queuePopulator/QueuePopulatorExtension');
+
+class LifecycleQueuePopulator extends QueuePopulatorExtension {
+
+    /**
+     * Pre-create the zookeeper node path for bucket lifecycle nodes, if
+     * necessary.
+     * @param {Function} cb - The callback function.
+     * @return {undefined}
+     */
+    _setup(cb) {
+        return this.zkClient.getData(this.zkPath, err => {
+            if (err) {
+                if (err.name !== 'NO_NODE') {
+                    this.log.error('Could not get lifecycle nodes', {
+                        method: 'LifecycleQueuePopulator._setup',
+                        error: err,
+                    });
+                    return cb(err);
+                }
+                return this.zkClient.mkdirp(this.zkPath, err => {
+                    if (err) {
+                        this.log.error('Could not create path in zookeeper', {
+                            method: 'LifecycleQueuePopulator._setup',
+                            zkPath: this.zkPath,
+                            error: err,
+                        });
+                        return cb(err);
+                    }
+                    return cb();
+                });
+            }
+            return cb();
+        });
+    }
+
+    /**
+     * Create a new zookeeper node for any bucket that has a lifecycle
+     * configuration. Remove the node if the bucket is deleted or the
+     * configuration is deleted.
+     * @param {Object} entry - The record log entry from metadata.
+     * @return {undefined}
+     */
+    _updateZookeeper(entry) {
+        const value = JSON.parse(entry.value);
+        const path = `${this.zkPath}/${value.owner}:${value.name}`;
+        return this.zkClient.exists(path, (err, stat) => {
+            if (err) {
+                this.log.error('Could not check existence of zookeeper node', {
+                    method: 'LifecycleQueuePopulator._updateZookeeper',
+                    zkPath: path,
+                    error: err,
+                });
+                return undefined;
+            }
+            // Remove existing node if deleting the bucket or its configuration.
+            if (stat &&
+            (value.deleted || value.lifecycleConfiguration === null)) {
```

Comment:
I am assuming `lifecycleConfiguration` should be `null` if nonexistent because it is a part of bucket info (https://github.com/scality/Arsenal/blob/master/lib/models/BucketInfo.js#L139). I am considering filtering based on the existence of this property in the `entry.value`, since the `usersBucket` is not defined when putting a lifecycle config. I've pushed those changes for your review.